internal/provider/organization_resource.go

```diff
@@ -223,6 +417,25 @@ func (r *OrganizationResource) Update(ctx context.Context, req resource.UpdateRe
 		"icon":         org.Icon,
 	})
 
+	tflog.Trace(ctx, "updating group sync", map[string]any{
+		"orgID": orgID,
+	})
+	if !data.GroupSync.IsNull() {
+		resp.Diagnostics.Append(r.patchGroupSync(ctx, orgID, data.GroupSync)...)
```

Comment:
nit: do the `IsNull` checks in each `patch*` function

Reply:
I like that this way makes it clear that we don't call those if nothing is set tho